### PR TITLE
Propagate all field values to formData for autocomplete

### PR DESCRIPTION
**Pull request by Mitch-A6**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)

## Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No

## Summary
- Bulk-filling fields with a browser's autocomplete may not fire Input/Change/Blur events.
- To reproduce this bug, Safari is the most consistent. Start the [10-10CG application](https://staging.va.gov/family-member-benefits/apply-for-caregiver-assistance-form-10-10cg/introduction) and proceed to the Veteran's home address page. Populate the fields from address autocomplete data (cmd-shift-A in Safari), and note that while this puts values in some fields, clicking Continue blanks them out, since the values were not added to the formData. 
- While addressing this for the Veteran is difficult due to the FormUI, we're able to fix it for the Primary Caregiver and both Secondary Caregivers by propagating all field values to the formData on any of their Change or Blur events.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/94536

## Testing done
- confirmed that we can autocomplete the Caregiver address fields, and the values are stored when continuing instead of being discarded on blur. we can see the final values on the Review page at the end.

## Screenshots
Before: 
Screenshot 2024-12-11 at 2 16 29 PM

After:
Screenshot 2024-12-11 at 2 16 21 PM

## What areas of the site does it impact?
- 10-10CG Caregiver Address pages

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/caregivers/components/FormFields/AddressWithAutofill.jsx [hunk 1]
```diff
@@ -82,10 +94,23 @@ const PrimaryAddressWithAutofill = props => {
...
   const handleBlur = useCallback(
     event => {
-      const fieldName = event.target.name.split('_').pop();
+      const { name } = event.target;
+
+      if (!name) {
+        // blurring the County's AddressCountyDescription component causes
+        // an error below when attempting the split. this avoids cascading the
+        // remaining blur behavior to it.
+        return;
```
Comment (Mitch-A6) [inline src/applications/caregivers/components/FormFields/AddressWithAutofill.jsx:103]: 🤷 This prevents the log message complaining about the `_undefined_.split` failure. I'm not sure if it's possible to cover this with our tests without `export`ing the `handleBlur` method or something else awkward? The failure doesn't cause a proper UI error or anything. Open to suggestions!

Comment (Mitch-A6) [reply src/applications/caregivers/components/FormFields/AddressWithAutofill.jsx:103]: Note this bug does manifest as a full-screen error, but only on localhost:

```
Uncaught runtime errors:
×
ERROR
Cannot read properties of undefined (reading 'split')
TypeError: Cannot read properties of undefined (reading 'split')
    at http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:62879:26
    at HTMLUnknownElement.callCallback (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164303:14)
    at Object.invokeGuardedCallbackDev (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164352:16)
    at invokeGuardedCallback (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164414:31)
    at invokeGuardedCallbackAndCatchFirstError (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164428:25)
    at executeDispatch (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168601:3)
    at processDispatchQueueItemsInOrder (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168633:7)
    at processDispatchQueue (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168646:5)
    at dispatchEventsForPlugins (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168657:3)
    at http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168866:12
ERROR
Cannot read properties of undefined (reading 'split')
TypeError: Cannot read properties of undefined (reading 'split')
    at http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:62879:26
    at HTMLUnknownElement.callCallback (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164303:14)
    at Object.invokeGuardedCallbackDev (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164352:16)
    at invokeGuardedCallback (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164414:31)
    at invokeGuardedCallbackAndCatchFirstError (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:164428:25)
    at executeDispatch (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168601:3)
    at processDispatchQueueItemsInOrder (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168633:7)
    at processDispatchQueue (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168646:5)
    at dispatchEventsForPlugins (http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168657:3)
    at http://localhost:3001/generated/1010cg-application-caregiver-assistance.entry.js:168866:12
    ```
Screenshot 2024-12-11 at 3 46 14 PM

Comment (coope93) [reply src/applications/caregivers/components/FormFields/AddressWithAutofill.jsx:103]: I took a crack at some specs for this and am also struggling to mimic the onBlur behavior for the component. I think the comment helps a lot with the context. Not sure if moving the component out of the input is a valid option or not, but that may cause the blur to not get triggered?

**Review by coope93 — DISMISSED** [review]

**Review by coope93 — DISMISSED** [review]

File: src/applications/caregivers/components/FormFields/AddressWithAutofill.jsx (hunk 1 above)
Comment (Mitch-A6) [reply src/applications/caregivers/components/FormFields/AddressWithAutofill.jsx]: UPDATE adding an `event.stopPropagation()` to the Description field fixed this nicely! Got a test in for it, too.

File: src/applications/caregivers/tests/unit/components/FormDescriptions/AddressCountyDescriptions.unit.spec.js [hunk 2]
```diff
@@ -1,22 +1,38 @@
...
 describe('CG <CaregiverCountyDescription>', () => {
-  it('should render `va-additional-info` component with neccessary attribute(s)', () => {
+  it('should render `va-additional-info` component with necessary attribute(s)', () => {
     const { container } = render(<CaregiverCountyDescription />);
     const selector = container.querySelector('va-additional-info');
     expect(selector).to.exist;
     expect(selector).to.have.attr('trigger');
   });
+
+  it('should prevent an outside blur event from executing', () => {
```
Comment (coope93) [inline src/applications/caregivers/tests/unit/components/FormDescriptions/AddressCountyDescriptions.unit.spec.js:18]: 👏

**Review by coope93 — APPROVED** [review]

**Pull request merged**